apsuite/optics_analysis/coupling_correction.py

```diff
@@ -0,0 +1,149 @@
+"""."""
+
+from copy import deepcopy as _dcopy
+import numpy as np
+from apsuite.commissioning_scripts.calc_orbcorr_mat import OrbRespmat
+import pyaccel
+
+
+class CouplingCorr():
```

Comment:
I think the class name or correction method should indicate the correction is based on reducing off-diagonal components of the orbit response matrix, since in the future new algorithms might be added using different methods.

Reply:
ok, I've changed the correction loop method name to indicate the correction method. Then I created a 'coupling_correction' method which take the correction method as input and chooses the corresponding function to run. So far it is only implemented the orbrespm based method.